### PR TITLE
Avoid putting in root namespace to generated vb settings file

### LINKED ISSUE #7102: Duplicate root namespace in settings file with sdk-style project
**Visual Studio Version**: 16.9.3

**Summary**:
When using an SDK-style project and vb.net, the automated tool generates a duplicate root namespace for settings files.

**Steps to Reproduce**:

1. Create a new SDK-style project for vb.net:

```xml
<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net472</TargetFramework>
    <UseWindowsForms>true</UseWindowsForms>
    <StartupObject>Test.Module1</StartupObject>
    <RootNamespace>Test</RootNamespace>
  </PropertyGroup>
  <ItemGroup>
    <Reference Include="mscorlib" />
  </ItemGroup>
  <ItemGroup>
    <Compile Update="MySettings.Designer.vb">
      <DesignTimeSharedInput>True</DesignTimeSharedInput>
      <AutoGen>True</AutoGen>
      <DependentUpon>MySettings.settings</DependentUpon>
    </Compile>
  </ItemGroup>
  <ItemGroup>
    <None Update="My Project\Settings.settings">
      <Generator>SettingsSingleFileGenerator</Generator>
    </None>
    <None Update="MySettings.settings">
      <Generator>SettingsSingleFileGenerator</Generator>
      <LastGenOutput>MySettings.Designer.vb</LastGenOutput>
    </None>
  </ItemGroup>
</Project>
```

2. Add a new settings file (test file is called `MySettings.settings`).
3. Inspect the generated `MySettings.Designer.vb` file.

**Expected Behavior**:

`MySettings.Designer.vb` should not wrap the code in the `RootNamespace` namespace, as that is automatically done by vb.net. This results in a duplicate namespace (i.e. `Test.Test.MySettings`).

This is what I expect the `MySettings.Designer.vb` file to look like:

image

**Actual Behavior**:

`MySettings.Designer.vb` has duplicated namespace:

image

**User Impact**:

Having to manually remove the namespace after saving the settings file or compile errors when referring to classes with their fully qualified name.

The correct code results when the `RootNamespace` is removed from the vb.net project but then all other classes fall out of the namespace. I'm pretty sure it doesn't do this when you use the legacy project style as the problem only showed up after switching to the SDK-style project.

### PR DESCRIPTION
**Pull request by adamint**
Fixes #7102
Running the settings generator now results in:
image

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/8322)

### CODE REVIEW

File: src/Microsoft.VisualStudio.Editors/SettingsDesigner/SettingsSingleFileGeneratorBase.vb [hunk 1]
```diff
@@ -251,7 +248,7 @@ Namespace Microsoft.VisualStudio.Editors.SettingsDesigner
 
             ' Create a new namespace to put our class in
             '
-            Dim ns As New CodeNamespace(DesignerFramework.DesignUtil.GenerateValidLanguageIndependentNamespace(DefaultNamespace))
+            Dim ns As New CodeNamespace()
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.Editors/SettingsDesigner/SettingsSingleFileGeneratorBase.vb:251]: Have you tested this change on C# projects too?

File: src/Microsoft.VisualStudio.Editors/SettingsDesigner/SettingsSingleFileGeneratorBase.vb [hunk 2]
```diff
@@ -570,17 +567,12 @@ Namespace Microsoft.VisualStudio.Editors.SettingsDesigner
         ''' Creates a string representation of the full-type name give the project's root-namespace, the default namespace
         ''' into which we are generating, and the name of the class
         ''' </summary>
-        ''' <param name="projectRootNamespace">project's root namespace (may be String.Empty)</param>
         ''' <param name="defaultNamespace">namespace into which we are generating (may be String.Empty)</param>
         ''' <param name="typeName">the type of the settings-class we are generating</param>
-        Private Shared Function GetFullTypeName(projectRootNamespace As String, defaultNamespace As String, typeName As String) As String
+        Private Shared Function GetFullTypeName(defaultNamespace As String, typeName As String) As String
```
Comment (ocallesp) [inline src/Microsoft.VisualStudio.Editors/SettingsDesigner/SettingsSingleFileGeneratorBase.vb:572]: Does this fixes the issue in VB ? does it work for C# ? the namespace might be coming from CPS.

if you hover the mouse over the class definition, does it show the correct namespace ?

File: src/Microsoft.VisualStudio.Editors/SettingsDesigner/SettingsSingleFileGeneratorBase.vb (hunk 1 above)
Comment (adamint) [reply src/Microsoft.VisualStudio.Editors/SettingsDesigner/SettingsSingleFileGeneratorBase.vb]: yep, see below response

File: src/Microsoft.VisualStudio.Editors/SettingsDesigner/SettingsSingleFileGeneratorBase.vb (hunk 2 above)
Comment (adamint) [reply src/Microsoft.VisualStudio.Editors/SettingsDesigner/SettingsSingleFileGeneratorBase.vb:581]: it fixes the issue in VB, still works for c#.

VB:
image

C#
image

**Comment by melytc** [comment]
In https://github.com/dotnet/project-system/pull/8355 I noticed this also happens:
image


Does your changes fix this issue, too?

File: src/Microsoft.VisualStudio.Editors/SettingsGlobalObject/SettingsGlobalObjectProvider.vb [hunk 3]
```diff
@@ -1298,7 +1298,10 @@ Namespace Microsoft.VisualStudio.Editors.SettingsGlobalObjects
 
             Dim isDesignTime As Boolean = True
 
-            Dim ccu As CodeCompileUnit = SettingsSingleFileGeneratorBase.Create(_hierarchy, designTimeSettingsToPresent, _namespace, fileName, isDesignTime, TypeAttributes.Public Or TypeAttributes.Sealed)
+            Dim isVB As Boolean = fileName.EndsWith(".vb")
+
+
```
Comment (tmeschter) [inline src/Microsoft.VisualStudio.Editors/SettingsGlobalObject/SettingsGlobalObjectProvider.vb:1303]: Nit: extra line

**Review by tmeschter — COMMENTED** [review]
Have you tested this with .NET Framework projects on the old project system?

**Comment by adamint** [comment]
> Have you tested this with .NET Framework projects on the old project system?

@tmeschter yes I have. This is the result of adding a new settings file in a .net framework winforms project: 
image
works as expected

**Review by melytc — APPROVED** [review]
Approving 🙂 thanks for validating the different scenarios.

**Review by drewnoakes — APPROVED** [review]

**Review by ocallesp — APPROVED** [review]

**Comment by ocallesp** [comment]
@Adamint what is blocking this merge ? If nothing is blocking it, then please merge it.

**Pull request merged**